### PR TITLE
fix: Allow null values in most places of an event

### PR DESCRIPTION
**Pull request by untitaker**
This is necessary for PII stripping to not cause validation errors

 - [x] fix arrays with null values

Files left:

- [x] `src/sentry/interfaces/contexts.py`
- [x] `src/sentry/interfaces/message.py`
- [x] `src/sentry/interfaces/http.py`
- [x] `src/sentry/interfaces/debug_meta.py`
- [x] `src/sentry/interfaces/geo.py`
- [x] `src/sentry/interfaces/template.py`
- [x] `src/sentry/interfaces/threads.py`
- [x] `src/sentry/interfaces/user.py`

less important for now afaik:

- [ ] `src/sentry/interfaces/query.py`
- [ ] `src/sentry/interfaces/security.py`
- [ ] `src/sentry/interfaces/repos.py`

does not apply:

- [ ] `src/sentry/interfaces/schemas.py`

#sync-getsentry

### CODE REVIEW

File: src/sentry/interfaces/sdk.py [hunk 1]
```diff
@@ -49,12 +49,7 @@ class Sdk(Interface):
     @classmethod
     def to_python(cls, data):
         name = data.get('name')
-        if not name:
-            raise InterfaceValidationError("No 'name' value")
-
         version = data.get('version')
-        if not version:
```
Comment (markstory) [inline src/sentry/interfaces/sdk.py]: Would an SDK not send its version because of PII stripping?

File: tests/sentry/interfaces/test_message.py [hunk 2]
```diff
@@ -19,6 +19,13 @@ def interface(self):
             )
         )
 
+    def test_null_values(self):
+        sink = {}
+        assert Message.to_python({}).to_json() == sink
+        assert Message.to_python({'message': None}).to_json() == sink
+        assert Message.to_python({'formatted': None}).to_json() == sink
+        assert Message.to_python({'params': None}).to_json() == sink
```
Comment (markstory) [inline tests/sentry/interfaces/test_message.py]: Do we know if downstream consumers of the interface types will be tolerant of previously required keys being missing?

File: src/sentry/interfaces/sdk.py (hunk 1 above)
Comment (untitaker) [reply src/sentry/interfaces/sdk.py]: yes exactly, relay might strip this value out

File: tests/sentry/interfaces/test_message.py (hunk 2 above)
Comment (untitaker) [reply tests/sentry/interfaces/test_message.py]: we're still trying to figure this out. We're testing this on garbage right now with mixed results

File: src/sentry/interfaces/debug_meta.py [hunk 3]
```diff
@@ -98,13 +98,19 @@ class DebugMeta(Interface):
 
     @classmethod
     def to_python(cls, data):
-        images = data.get('images', [])
-        is_debug_build = data.get('is_debug_build')
+        is_debug_build = data.get('is_debug_build', None)
```
[the remark this answers is not in the document]
Comment (untitaker) [reply src/sentry/interfaces/debug_meta.py]: yeah probably. We might have to split this PR up anyway because it's super high-risk to deploy at once

**Review by mitsuhiko — APPROVED** [review]
let's roll

**Comment by bretthoerner** [comment]
Woo boy super curious to see what untested edges this finds. :)

**Pull request merged**